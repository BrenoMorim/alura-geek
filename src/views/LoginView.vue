<template>
  <form class="formulario pagina-formulario" @submit.prevent="fazerLogin()">
    <h3 class="formulario__titulo formulario__titulo--maior ">Iniciar Sessão</h3>
    <input v-model="usuario.email" name="email" type="email" class="formulario__campo formulario__campo--maior" placeholder="Digite seu email" required>
    <input v-model="usuario.senha" type="password" name="senha" class="formulario__campo formulario__campo--maior" placeholder="Digite sua senha" required>
    <button class="formulario__botao formulario__botao--maior" type="submit">Entrar</button>
    <p v-if="mensagemDeErro" class="mensagem-erro">{{mensagemDeErro}}</p>
  </form>
</template>

<script lang="ts">
import store from "@/store";
import { FAZER_LOGIN } from "@/types/Actions";
import { defineComponent } from "@vue/runtime-core";

export default defineComponent({
    name: 'LoginView',
    data() {
      return {
        usuario: { email: "", senha: ""},
        mensagemDeErro: ''
      }
    },
    methods: {
      async fazerLogin() {
          const loginRealizado = await store.dispatch(FAZER_LOGIN, this.usuario);
          if (loginRealizado) {
            this.$router.push({ name: 'home'});
            this.mensagemDeErro = "";
            return;
          }
          this.mensagemDeErro = "Email ou senha inválidos!";
      }
    }
})
</script>

<style scoped>
@import url('@/assets/css/formulario.css');
</style>