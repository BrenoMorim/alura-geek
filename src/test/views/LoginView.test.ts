import { expect, describe, it, vi } from "vitest";
import { mount } from '@vue/test-utils';
import LoginView from '../../views/LoginView.vue';
import store from "../../store";

describe('Na tela de login', () => {
    
    const wrapper = mount(LoginView);
    const formulario = wrapper.get('.formulario');
    const inputEmail = formulario.get('#email');
    const inputSenha = formulario.get('#senha');
    const botaoSubmit = formulario.get('.formulario__botao');

    it('Deve ser renderizada corretamente', () => {
        expect(wrapper.isVisible()).toBeTruthy();
        expect(wrapper).toMatchSnapshot();
    });

    it('Deve conter formulário com título, inputs e botão', () => {
        const titulo = formulario.get('.formulario__titulo');
        expect(titulo.text()).toBe('Iniciar Sessão');

        expect(inputEmail.attributes('placeholder')).toBe('Digite seu email');
        expect(inputSenha.attributes('placeholder')).toBe('Digite sua senha');
        expect(botaoSubmit.text()).toBe('Entrar');
        expect(botaoSubmit.attributes('disabled')).toBeFalsy();
    });

    it('Deve tentar fazer login e mostrar mensagem de erro quando não for possível logar', async () => {
        const mockFazerLogin = vi.fn().mockImplementation(() => false);
        store.dispatch = mockFazerLogin;

        await inputEmail.setValue('email@valido.com');
        await inputSenha.setValue('senhaCorreta');
        await botaoSubmit.trigger('click');

        // Esperando a mensagem aparecer e mock ser chamado
        setTimeout(() => {
            const mensagemErro = formulario.get('.mensagem-erro');
            expect(mensagemErro.text()).toBe('Email ou senha inválidos!');
            
            expect(mockFazerLogin).toHaveBeenCalledOnce();
        }, 200);
    });
});
