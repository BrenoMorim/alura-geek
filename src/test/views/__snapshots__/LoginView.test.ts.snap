// Vitest Snapshot v1

exports[`Na tela de login > Deve ser renderizada corretamente 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <form
        class="formulario pagina-formulario"
        data-v-45f5edd7=""
      >
        <h3
          class="formulario__titulo formulario__titulo--maior"
          data-v-45f5edd7=""
        >
          Iniciar Sess√£o
        </h3>
        <input
          class="formulario__campo formulario__campo--maior"
          data-v-45f5edd7=""
          id="email"
          name="email"
          placeholder="Digite seu email"
          required=""
          type="email"
        />
        <input
          class="formulario__campo formulario__campo--maior"
          data-v-45f5edd7=""
          id="senha"
          name="senha"
          placeholder="Digite sua senha"
          required=""
          type="password"
        />
        <button
          class="formulario__botao formulario__botao--maior"
          data-v-45f5edd7=""
          type="submit"
        >
          Entrar
        </button>
        <!--v-if-->
      </form>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {},
      "da": null,
      "data": {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {},
      "refs": {
        "VTU_COMPONENT": {
          "fazerLogin": [Function],
          "hasOwnProperty": [Function],
          "mensagemDeErro": "",
          "usuario": {
            "email": "",
            "senha": "",
          },
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": [],
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": {},
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "constructor": 0,
            "fazerLogin": 4,
            "hasOwnProperty": 4,
            "mensagemDeErro": 2,
            "toJSON": 0,
            "usuario": 2,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "fazerLogin": [Function],
            "hasOwnProperty": [Function],
            "mensagemDeErro": "",
            "usuario": {
              "email": "",
              "senha": "",
            },
          },
          "da": null,
          "data": {
            "mensagemDeErro": "",
            "usuario": {
              "email": "",
              "senha": "",
            },
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {},
          "propsDefaults": {},
          "propsOptions": [
            {},
            [],
          ],
          "provides": {},
          "proxy": {
            "fazerLogin": [Function],
            "hasOwnProperty": [Function],
            "mensagemDeErro": "",
            "usuario": {
              "email": "",
              "senha": "",
            },
          },
          "refs": {},
          "render": [Function],
          "renderCache": [
            [Function],
            [Function],
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": [],
            "effects": [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": {},
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Iniciar Sess√£o",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <h3
                  class="formulario__titulo formulario__titulo--maior"
                  data-v-45f5edd7=""
                >
                  Iniciar Sess√£o
                </h3>,
                "key": null,
                "patchFlag": -1,
                "props": {
                  "class": "formulario__titulo formulario__titulo--maior",
                },
                "ref": null,
                "scopeId": "data-v-45f5edd7",
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "h3",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": [
                  {
                    "arg": undefined,
                    "dir": {
                      "beforeUpdate": [Function],
                      "created": [Function],
                      "mounted": [Function],
                    },
                    "instance": {
                      "fazerLogin": [Function],
                      "hasOwnProperty": [Function],
                      "mensagemDeErro": "",
                      "usuario": {
                        "email": "",
                        "senha": "",
                      },
                    },
                    "modifiers": {},
                    "oldValue": undefined,
                    "value": "",
                  },
                ],
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <input
                  class="formulario__campo formulario__campo--maior"
                  data-v-45f5edd7=""
                  id="email"
                  name="email"
                  placeholder="Digite seu email"
                  required=""
                  type="email"
                />,
                "key": null,
                "patchFlag": 512,
                "props": {
                  "class": "formulario__campo formulario__campo--maior",
                  "id": "email",
                  "name": "email",
                  "onUpdate:modelValue": [Function],
                  "placeholder": "Digite seu email",
                  "required": "",
                  "type": "email",
                },
                "ref": null,
                "scopeId": "data-v-45f5edd7",
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": [
                  {
                    "arg": undefined,
                    "dir": {
                      "beforeUpdate": [Function],
                      "created": [Function],
                      "mounted": [Function],
                    },
                    "instance": {
                      "fazerLogin": [Function],
                      "hasOwnProperty": [Function],
                      "mensagemDeErro": "",
                      "usuario": {
                        "email": "",
                        "senha": "",
                      },
                    },
                    "modifiers": {},
                    "oldValue": undefined,
                    "value": "",
                  },
                ],
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <input
                  class="formulario__campo formulario__campo--maior"
                  data-v-45f5edd7=""
                  id="senha"
                  name="senha"
                  placeholder="Digite sua senha"
                  required=""
                  type="password"
                />,
                "key": null,
                "patchFlag": 512,
                "props": {
                  "class": "formulario__campo formulario__campo--maior",
                  "id": "senha",
                  "name": "senha",
                  "onUpdate:modelValue": [Function],
                  "placeholder": "Digite sua senha",
                  "required": "",
                  "type": "password",
                },
                "ref": null,
                "scopeId": "data-v-45f5edd7",
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Entrar",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <button
                  class="formulario__botao formulario__botao--maior"
                  data-v-45f5edd7=""
                  type="submit"
                >
                  Entrar
                </button>,
                "key": null,
                "patchFlag": -1,
                "props": {
                  "class": "formulario__botao formulario__botao--maior",
                  "type": "submit",
                },
                "ref": null,
                "scopeId": "data-v-45f5edd7",
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "button",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "v-if",
                "component": null,
                "dirs": null,
                "dynamicChildren": [],
                "dynamicProps": null,
                "el": <!--v-if-->,
                "key": null,
                "patchFlag": 0,
                "props": null,
                "ref": null,
                "scopeId": "data-v-45f5edd7",
                "shapeFlag": 8,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(Comment),
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": [
                  {
                    "arg": undefined,
                    "dir": {
                      "beforeUpdate": [Function],
                      "created": [Function],
                      "mounted": [Function],
                    },
                    "instance": {
                      "fazerLogin": [Function],
                      "hasOwnProperty": [Function],
                      "mensagemDeErro": "",
                      "usuario": {
                        "email": "",
                        "senha": "",
                      },
                    },
                    "modifiers": {},
                    "oldValue": undefined,
                    "value": "",
                  },
                ],
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <input
                  class="formulario__campo formulario__campo--maior"
                  data-v-45f5edd7=""
                  id="email"
                  name="email"
                  placeholder="Digite seu email"
                  required=""
                  type="email"
                />,
                "key": null,
                "patchFlag": 512,
                "props": {
                  "class": "formulario__campo formulario__campo--maior",
                  "id": "email",
                  "name": "email",
                  "onUpdate:modelValue": [Function],
                  "placeholder": "Digite seu email",
                  "required": "",
                  "type": "email",
                },
                "ref": null,
                "scopeId": "data-v-45f5edd7",
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": [
                  {
                    "arg": undefined,
                    "dir": {
                      "beforeUpdate": [Function],
                      "created": [Function],
                      "mounted": [Function],
                    },
                    "instance": {
                      "fazerLogin": [Function],
                      "hasOwnProperty": [Function],
                      "mensagemDeErro": "",
                      "usuario": {
                        "email": "",
                        "senha": "",
                      },
                    },
                    "modifiers": {},
                    "oldValue": undefined,
                    "value": "",
                  },
                ],
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <input
                  class="formulario__campo formulario__campo--maior"
                  data-v-45f5edd7=""
                  id="senha"
                  name="senha"
                  placeholder="Digite sua senha"
                  required=""
                  type="password"
                />,
                "key": null,
                "patchFlag": 512,
                "props": {
                  "class": "formulario__campo formulario__campo--maior",
                  "id": "senha",
                  "name": "senha",
                  "onUpdate:modelValue": [Function],
                  "placeholder": "Digite sua senha",
                  "required": "",
                  "type": "password",
                },
                "ref": null,
                "scopeId": "data-v-45f5edd7",
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "v-if",
                "component": null,
                "dirs": null,
                "dynamicChildren": [],
                "dynamicProps": null,
                "el": <!--v-if-->,
                "key": null,
                "patchFlag": 0,
                "props": null,
                "ref": null,
                "scopeId": "data-v-45f5edd7",
                "shapeFlag": 8,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(Comment),
              },
            ],
            "dynamicProps": null,
            "el": <form
              class="formulario pagina-formulario"
              data-v-45f5edd7=""
            >
              <h3
                class="formulario__titulo formulario__titulo--maior"
                data-v-45f5edd7=""
              >
                Iniciar Sess√£o
              </h3>
              <input
                class="formulario__campo formulario__campo--maior"
                data-v-45f5edd7=""
                id="email"
                name="email"
                placeholder="Digite seu email"
                required=""
                type="email"
              />
              <input
                class="formulario__campo formulario__campo--maior"
                data-v-45f5edd7=""
                id="senha"
                name="senha"
                placeholder="Digite sua senha"
                required=""
                type="password"
              />
              <button
                class="formulario__botao formulario__botao--maior"
                data-v-45f5edd7=""
                type="submit"
              >
                Entrar
              </button>
              <!--v-if-->
            </form>,
            "key": null,
            "patchFlag": 32,
            "props": {
              "class": "formulario pagina-formulario",
              "onSubmit": [Function],
            },
            "ref": null,
            "scopeId": "data-v-45f5edd7",
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "form",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/home/breno/Studies/front-end/alura-geek/src/views/LoginView.vue",
            "__scopeId": "data-v-45f5edd7",
            "components": {},
            "data": [Function],
            "methods": {
              "fazerLogin": [Function],
            },
            "name": "LoginView",
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <form
          class="formulario pagina-formulario"
          data-v-45f5edd7=""
        >
          <h3
            class="formulario__titulo formulario__titulo--maior"
            data-v-45f5edd7=""
          >
            Iniciar Sess√£o
          </h3>
          <input
            class="formulario__campo formulario__campo--maior"
            data-v-45f5edd7=""
            id="email"
            name="email"
            placeholder="Digite seu email"
            required=""
            type="email"
          />
          <input
            class="formulario__campo formulario__campo--maior"
            data-v-45f5edd7=""
            id="senha"
            name="senha"
            placeholder="Digite sua senha"
            required=""
            type="password"
          />
          <button
            class="formulario__botao formulario__botao--maior"
            data-v-45f5edd7=""
            type="submit"
          >
            Entrar
          </button>
          <!--v-if-->
        </form>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "/home/breno/Studies/front-end/alura-geek/src/views/LoginView.vue",
          "__scopeId": "data-v-45f5edd7",
          "components": {},
          "data": [Function],
          "methods": {
            "fazerLogin": [Function],
          },
          "name": "LoginView",
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <form
          class="formulario pagina-formulario"
          data-v-45f5edd7=""
        >
          <h3
            class="formulario__titulo formulario__titulo--maior"
            data-v-45f5edd7=""
          >
            Iniciar Sess√£o
          </h3>
          <input
            class="formulario__campo formulario__campo--maior"
            data-v-45f5edd7=""
            id="email"
            name="email"
            placeholder="Digite seu email"
            required=""
            type="email"
          />
          <input
            class="formulario__campo formulario__campo--maior"
            data-v-45f5edd7=""
            id="senha"
            name="senha"
            placeholder="Digite sua senha"
            required=""
            type="password"
          />
          <button
            class="formulario__botao formulario__botao--maior"
            data-v-45f5edd7=""
            type="submit"
          >
            Entrar
          </button>
          <!--v-if-->
        </form>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.37",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": {
    "fazerLogin": [Function],
    "hasOwnProperty": [Function],
    "mensagemDeErro": "",
    "usuario": {
      "email": "",
      "senha": "",
    },
  },
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": <form
    class="formulario pagina-formulario"
    data-v-45f5edd7=""
  >
    <h3
      class="formulario__titulo formulario__titulo--maior"
      data-v-45f5edd7=""
    >
      Iniciar Sess√£o
    </h3>
    <input
      class="formulario__campo formulario__campo--maior"
      data-v-45f5edd7=""
      id="email"
      name="email"
      placeholder="Digite seu email"
      required=""
      type="email"
    />
    <input
      class="formulario__campo formulario__campo--maior"
      data-v-45f5edd7=""
      id="senha"
      name="senha"
      placeholder="Digite sua senha"
      required=""
      type="password"
    />
    <button
      class="formulario__botao formulario__botao--maior"
      data-v-45f5edd7=""
      type="submit"
    >
      Entrar
    </button>
    <!--v-if-->
  </form>,
}
`;
